addons/mrp/migrations/11.0.2.0/post-migration.py

```diff
@@ -0,0 +1,207 @@
+# Copyright 2018 Eficent <http://www.eficent.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+from psycopg2.extensions import AsIs
+
+
+def fill_mrp_document(cr):
```

Comment:
I think you don't need to create this, as it's automatically populated without this code. This is needed in case you have versioned files, which is a new feature in v11. Can you please try without doing this?

Reply:
It's needed. In v10, the "mrp_documents" are put together with other kind of attachments in ir_attachment. But in v11, they separated this specific case. Thus, this script what does is to put this kind of mrp attachments in this specific table. It's not automatically populated, because this script keeps the active and priority fields, otherwise they are lost.

![workspace 1_045](https://user-images.githubusercontent.com/25005517/49796937-4b6c7980-fd3e-11e8-9dea-b44ac145736f.png)

Reply:
OK. I think it can be omitted if active and priority have default values then.

Reply:
priority doesn't have default

Reply:
Then the default value will be 0 or NULL ;), but do you get my point?

Reply:
I really don't understand your cryptic point. My point is that this script is absolute necessary because if not put, then the document attached is lost.

It's very easy to test:
1. create a bill of materials, add a component, and in this component click to add an attachment as shown in the above image, and then upload a file in the new attachment.
2. then migrate.
3. if the migration doesn't have this script, then you will find in the bill of materials, in the component you attached the file, you click on the attachment icon and you won't find the uploaded document.

Reply:
> - When using that button that you are showing me, in v10 it creates a simple ir.attachment.

Yeah, and in v11 also creates an ir.attachment but also a mrp.document. In fact, creating a mrp.document it also creates an ir.attachment because mrp.document model it's inheriting from ir.attachment. So the script it's just creating the corresponding mrp.document records that are linked with those ir.attachment, like if they were created in v11 instead. And uses the following criteria: there should exist a mrp.document for each product attachment, filtering the products by which exists a bom line and excluding those that doesn't exist in any bom line, mimicking the way mrp.document are created in v11.

> So my point is to not populate unneeded mrp.document records if the needed fields haven't changed in the linked ir.attachment, so you might reduce the inserted records from thousands to barely zero.

What I understand from this words is that you are ok with the script but that you want it filter a bit more the case in which the mrp.document are created. But I don't understand your criteria of filtering. Your criteria is that the script only should create mrp.document if the active and priority fields are different from their default, because your point it's that they are automatically populated. But that point is false, the mrp.document doesn't automatically populate, and thus, your criteria is wrong. Please, check my test I put in my previous comment. I think it's very important to be able to access to the attached document through clicking that attachment icon 📓 .

Reply:
Well, maybe both are right. Look this short video:

![peek 12-12-2018 21-40](https://user-images.githubusercontent.com/7165771/49897519-b3ae7e80-fe56-11e8-813d-2756240d6e44.gif)

Here I introduce the attachment in the product, and you can see that the icon changes, but when you click, there isn't anything. In v10, this works.

So, let's continue with this for now, and we should notify a bug about this with 2 options:

1. Change compute field that decides the icon in the BoM line to only change when there are mrp.document records.
2. View also ir.attachment records.

I think second point is harder to be fixed due to the new structure, so guessing that, this query is correct as is.